linkage-monitor/src/test/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitorTest.java

```diff
@@ -60,20 +61,47 @@ public void testFindSnapshotVersion() throws VersionRangeResolutionException {
   @Test
   public void testBomSnapshot()
       throws VersionRangeResolutionException, InvalidVersionSpecificationException {
-    VersionRangeResult dummyVersionRangeResult = new VersionRangeResult(new VersionRangeRequest());
+    VersionRangeResult versionWithSnapshot = new VersionRangeResult(new VersionRangeRequest());
+    VersionRangeResult versionWithoutSnapshot = new VersionRangeResult(new VersionRangeRequest());
     GenericVersionScheme versionScheme = new GenericVersionScheme();
-    dummyVersionRangeResult.setVersions(
+    versionWithSnapshot.setVersions(
         ImmutableList.of(
-            versionScheme.parseVersion("1.2.3"),
-            versionScheme.parseVersion("2.1.1"),
-            versionScheme.parseVersion("2.1.2-SNAPSHOT")));
+            versionScheme.parseVersion("3.6.0"),
+            versionScheme.parseVersion("3.7.0"),
+            versionScheme.parseVersion("3.8.0-SNAPSHOT")));
+    versionWithoutSnapshot.setVersions(
+        ImmutableList.of(versionScheme.parseVersion("1.2.3"), versionScheme.parseVersion("1.1.1")));
 
     RepositorySystem mockSystem = mock(RepositorySystem.class);
     when(mockSystem.resolveVersionRange(
             any(RepositorySystemSession.class), any(VersionRangeRequest.class)))
-        .thenReturn(dummyVersionRangeResult);
+        .thenReturn(versionWithoutSnapshot); // other invocations than protobuf-java
+    when(mockSystem.resolveVersionRange(
+            any(RepositorySystemSession.class),
+            argThat(request -> "protobuf-java".equals(request.getArtifact().getArtifactId()))))
+        .thenReturn(versionWithSnapshot); // invocation for protobuf-java
 
     Bom snapshotBom = LinkageMonitor.copyWithSnapshot(mockSystem, bom);
-    assertEquals("2.1.2-SNAPSHOT", snapshotBom.getManagedDependencies().get(0).getVersion());
+
+    assertEquals(
+        "The first element of the SNAPSHOT BOM should be the same as the original BOM",
+        "protobuf-java",
+        snapshotBom.getManagedDependencies().get(0).getArtifactId());
+    assertEquals(
+        "The protobuf-java artifact should have SNAPSHOT version",
+        "3.8.0-SNAPSHOT",
+        snapshotBom.getManagedDependencies().get(0).getVersion());
+
+    int bomSize = bom.getManagedDependencies().size();
+    assertEquals(
+        "Snapshot BOM should have the same length as original BOM.",
+        bomSize,
+        snapshotBom.getManagedDependencies().size());
+    for (int i = 1; i < bomSize; ++i) {
+      assertEquals(
+          "The other artifacts than protobuf-java should have the original version",
```

Comment:
"The other artifacts than" --> "Artifacts other than"

Reply:
Updated